packages/flutter/lib/src/material/material_app.dart

```diff
@@ -106,23 +122,32 @@ class _MaterialAppState extends State<MaterialApp> implements BindingObserver {
   }
 
   Widget build(BuildContext context) {
+    if (config.onLocaleChanged != null && _localeData == null) {
+      // If the app expects a locale but we don't yet know the locale, then
+      // don't build the widgets now.
+      return new Container();
```

Comment:
The initializeMessages function generated by the Dart Intl tools is async (probably because it might load a deferred library).  So config.onLocaleChanged had to be async as well.

In the framework, is there a good way to do initialization of things like the locale before starting the app?